docs/sources/mimir/operators-guide/architecture/query-sharding/index.md

```diff
@@ -139,6 +139,23 @@ parallel by the query-frontend, multiplying the previously set value of
 `-querier.max-query-parallelism` by
 `-query-frontend.query-sharding-total-shards`.
 
+## Cardinality estimation for query sharding (experimental)
+
+Since a high number of parallel sharded queries increases load on queriers and their dependencies, it is desirable to shard queries only as much as necessary.
+Queries that return more series, such as those that are of high cardinality, need to fetch more data and should therefore be split into a larger batch of shards.
+Queries that return few or no series should be executed with fewer or no shards at all.
+When determining the number of shards to use for a given query, the sharding logic can optionally take into account the cardinality observed during previous executions of the same query for similar time ranges.
+
+To enable this experimental feature, set `-query-frontend.query-sharding-max-series-per-shard` to a value representing roughly how many series each shard should fetch, and enable result caching (the result cache is used to store cardinality estimates).
```

Comment:
I'm wondering if we should rename the param to `-query-frontend.query-sharding-recommended-series-per-shard` (or something similar). Now that I see the doc and the confusion generated to Oleg too, I think calling it "max" is not much clear because looks like an upper bound, while what we really configure is just an "hint".

Reply:
@colega's suggestion sounds like a nice way to phrase it to me. @pracucci wdyt?

Reply:
I like it!